lib/collection.js

```diff
@@ -454,13 +473,15 @@ Collection.prototype.insertOne = function(doc, options, callback) {
   if (typeof options === 'function') (callback = options), (options = {});
   options = options || {};
 
-  // Add ignoreUndfined
+  // Add ignoreUndefined
   if (this.s.options.ignoreUndefined) {
     options = Object.assign({}, options);
     options.ignoreUndefined = this.s.options.ignoreUndefined;
```

Comment:
This should move to the validator, which it will when I implement the default portion of the validator as part of [NODE-1664](https://jira.mongodb.org/browse/NODE-1664). I want this PR to focus on the type checking only.

Reply:
:+1: